Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+
+We will use API Gateway and Lambda to create 6 endpoints (GetExpense, Add/UpdateExpense, GetTaggedExpense, GetBudget, Add/UpdateBudget, BudgetStatus) that will handle the creation, update, and retrieval of expense and budget information to satisfy our requirements.
+
+We will store expense and budget information in a table in DynamoDB.
+
+Amazon Elite Budget Service will also provide a web interface for users. A main page providing a variety of options to add, update and view expense and budget information.
+
+
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ExpenseModel
+
+String expenseId;
+String expenseName;
+String expenseAmount; 
+String tag;
+ZonedDateTime expenseDate;
+
+// BudgetModel
+
+String budgetId;
+String targetAmount; 
+Boolean status; 
+ZonedDateTime date;
+
+```
+
+## 6.2. _GetExpense Endpoint_
+
+* Accepts `GET` requests to `/expense/:date`
+* Accepts an expenseID and returns the corresponding ExpenseModel.
+    * If the given expenseID is not found, will throw a
+      `ExpenditureNotFoundException`
+
+## 6.3 _GetAllExpenses Endpoint_
+
+* Accepts `GET` requests to `/expense`
+* Returns all the expenses with any tag in the ExpenseModel format.
+    * If there is no data found, will throw a
+      `NoDataFoundException`
+
+## 6.4. GetTaggedExpense Endpoint
+
+* Accepts `GET` requests to `/expense/:tag`
+* Returns all the expenses for the requested tag in the ExpenseModel format.
+    * If there is no department found, will throw a
+      `InvalidDepartmentException`
+
+## 6.5. _UpdateExpense Endpoint_
+
+* Accepts `PUT` requests to `/expense/:date`
+* Accepts data to update an expense including an updated
+  expenseName, expenseAmount, tag, and date. Returns the updated expense.
+* If the expenseID or name is not found, will throw a   
+  `ExpenseNotFoundException`
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.6. _AddExpense Endpoint_
+
+* Accepts `POST` requests to `/expense/:date`
+* Accepts data to create a new expenditure which includes the
+  expenseId, expenseName, expenseAmount, tag, and date. Returns the new   
+  expense.
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.7. _GetBudget Endpoint_
+
+* Accepts `GET` requests to `/budget/:date`
+* Accepts a date and returns the corresponding BudgetModel.
+    * If the given date is not found, will throw a
+      `BudgetNotFoundException`
+
+## 6.6. _AddBudget Endpoint_
+
+* Accepts `POST` requests to `/budget/:date`
```

Comment:
I'm not sure if `:date` is the right parameter here (and throughout) when creating/modifying a budget? Let's discuss!